### ISSUE TITLE
How do I upgrade the firmware?

### ISSUE THREAD

**Issue by robotaro** [makerbase-motor/MKS-SERVO57D#17]
I have a MKS Servo 42C and I'd like to upgrade to v1.0.5, that has the real-time position control update, or newer.

I have no clue where to find that firmware or how to flash it to the controller. Does anyone have any clue?

**Comment by gjyskyblue** [comment]
An offline downloader is required to upgrade the firmware. If you need it, please contact the seller.   https://detail.tmall.com/item.htm?_u=k46rb8tc8ae&id=676740811159&spm=a1z09.2.0.0.1e0a2e8das3phD

**Comment by robotaro** [comment]
Hi gjyskyblue, thank you for the link! I'll see if I can order one. I've found this generic STM32 programmer (https://www.amazon.co.uk/Youmile-Emulator-Downloader-Programmer-Programming/dp/B07QBLNDPM/ref=sr_1_4?nsdOptOutParam=true&sr=8-4) but I don't know if it is compatible. 

Do you have experience in upgrading the firmware on these boards using the downloader you provided? If so, how did you get the newer firmware versions?

I have tried reaching out to MKSBase for the firmware but they never replied. If you know anything about where I can get firmware 1.0.5 I would be very grateful.

**Comment by daxliniere** [comment]
Yes please @makerbase-motor , where can we download the latest firmware, what hardware do we need and where is a step-by-step tutorial?
